shielder/contract/merkle.rs

```diff
@@ -0,0 +1,65 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use ink::{storage::Mapping, env::hash::{Sha2x256, CryptoHash}};
+
+use crate::{types::{Scalar, Set}, errors::ShielderError};
+
+
+pub const DEPTH: usize = 10;
+
+#[ink::storage_item]
+#[derive(Debug)]
+pub struct MerkleTree {
+    nodes: Mapping<u32, Scalar>,
+    roots_log: Set<Scalar>,
+    next_leaf_id: u32,
+    sz: u32,
```

Comment:
better to spell it out: `size` :)

Reply:
Also please add a comment on what is this exactly

Reply:
changed to size and commented merkle